### PR TITLE
Lost Mountains (New map for CTF)

### PR DESCRIPTION
**Pull request by LuismiInTheZone**
I lost old map so I needed to start again... I made something different with the help of the tips of olliy, Xenon, HobbitPower and bloopy, so thanks to you for helping me a bit how to use map creator and how to save it.

Sorry for all the problems it's my new map creating all right and I'm not expert with this.

### CODE REVIEW

**Comment by HobbitPower** [comment]
* This is not the full review, but i took a first look: The "cube" with the outer barriers still got some mistakes. The bottom and top layer is completely missing. So i could just dig down or stack up and then there is the "void". Also the outer barriers are not completely made out of the barrier glass/barrier stone.  i found default:stone, default:sand and some others. so i could just dig my way out of the barrier there. 
* There are no chests or i found none. You probably forgot to determine the chest zones. You can simply add the whole map as one chest zone i guess.
* There are 16 mese and 16 diamonds, but not a single iron ore. Also you can't mine them since we got no steel pick as a start item + you cant craft a pick without iron + you cant find any since there are no chests. Would add some iron, fits to the map imo and i would also add a steel pick in the initial_stuff

**Comment by LuismiInTheZone** [comment]
Everything it's suppose to be fixed... You can test it now and see if it's all ok, then votation for the map...

**Comment by HobbitPower** [comment]
> Everything it's suppose to be fixed... You can test it now and see if it's all ok, then votation for the map...

Okay, seems to work fine now. Found no (new) bugs yet. Before reading, i would wait for a whole review from map managers. The following are mostly personal preference. So no need to change your map again immediately ;)
* Some parts of the outer barriers look a bit weird, because there are random squares of ctf_map:stone. but that would be just a discussion about aesthetics. same for the red barrier. some of it goes into the ground so after the round starts, there are holes sometimes. you could fill these with "default:dirt_with_coniferous_litter" when there is the same right and left to it. afaik there is no problem having some digable blocks in the middle (where the red barrier is located), as long as the team zones are working properly. at it seems they are working just fine. So for me thats no major problem, its again a discussion about aesthetics.
* Also this is just a personal opinion, i am still in favour of adding a steel pick as an initial item. I mean you need it to dig the mese/diamond and for digging tunnels for example. Ofc now you can at least find picks in chests. Speaking of which: tried a few rounds and sometimes it took me a while to find chests. maybe the amount of chests are not enough for the size of the map, some players may complain about this, but thats your choice. Imho it would be still good to have some iron ores in the map (there are zero in the whole map). but yeah again ,just wait for more reviews.
* Ask a map manager about the licence. Idk if "CC-BY" is enough.

**Comment by NayiemWillems** [comment]
I just played this map. I like it :)

File: lost_mountains/map.conf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+phys_jump = 1
+phys_speed = 1
+size = return {["y"] = 152, ["x"] = 231, ["z"] = 266}
+map_version = 2
+enabled = true
+name = Lost Mountains
+phys_gravity = 1
+initial_stuff = return {}
+author = Luis_Mi
+hint = Look into the caves near your base, you'll find ores!
```
Comment (MoreKat) [inline lost_mountains/map.conf:14]: Better: 'Look in the dungeons near your base to find ores!'
During my review I was looking for actual caves, but the ores are in the dungeons.

File: lost_mountains/map.conf [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+teams = return {["red"] = {["pos2"] = {["y"] = 152, ["x"] = 117, ["z"] = 266}, ["enabled"] = true, ["pos1"] = {["y"] = -1, ["x"] = 231, ["z"] = 0}, ["flag_pos"] = {["y"] = 32, ["x"] = 193, ["z"] = 183}}, ["blue"] = {["pos2"] = {["y"] = 152, ["x"] = 117, ["z"] = 0}, ["enabled"] = true, ["pos1"] = {["y"] = 1, ["x"] = 0, ["z"] = 266}, ["flag_pos"] = {["y"] = 33, ["x"] = 33, ["z"] = 140}}}
+chests = return {{["pos2"] = {["y"] = 152, ["x"] = 231, ["z"] = 266}, ["pos1"] = {["y"] = -1, ["x"] = 117, ["z"] = 0}, ["amount"] = 25}, {["pos2"] = {["y"] = 152, ["x"] = 117, ["z"] = 266}, ["pos1"] = {["y"] = 1, ["x"] = 0, ["z"] = 0}, ["amount"] = 25}}
+phys_jump = 1
+phys_speed = 1
+size = return {["y"] = 152, ["x"] = 231, ["z"] = 266}
+map_version = 2
+enabled = true
+name = Lost Mountains
+phys_gravity = 1
+initial_stuff = return {}
```
Comment (MoreKat) [inline lost_mountains/map.conf:12]: I suggest giving everyone a pickaxe (mese or steel), ladders and torches

**Review by MoreKat — COMMENTED** [review]
Interesting map, looks cool. I'm not sure how enjoyable it is to climb mountains all day but we'll find out when it is played. I think you should add more ores to the dungeons, maybe not mese or diamonds but at least iron and coal (which ore is up to you though). I also suggest adding a full set of ladders (so no missing pieces like it is now) to the dungeon of the red base so it's as accessible as the one near the blue base. 

Definitely add some items to initial_stuff. I suggested a pickaxe to mine the ores and also for digging tunnels _through_ the mountain. Torches always come in handy, especially since this is a rather dark map because of the skybox. Ladders to facilitate climbing the mountains.

EDIT: It could be a fun idea to slightly up the phys_speed, to 1.3 or 1.4 or so.

**Comment by LuismiInTheZone** [comment]
> Interesting map, looks cool. I'm not sure how enjoyable it is to climb mountains all day but we'll find out when it is played. I think you should add more ores to the dungeons, maybe not mese or diamonds but at least iron and coal (which ore is up to you though). I also suggest adding a full set of ladders (so no missing pieces like it is now) to the dungeon of the red base so it's as accessible as the one near the blue base.
> 
> Definitely add some items to initial_stuff. I suggested a pickaxe to mine the ores and also for digging tunnels _through_ the mountain. Torches always come in handy, especially since this is a rather dark map because of the skybox. Ladders to facilitate climbing the mountains.
> 
> EDIT: It could be a fun idea to slightly up the phys_speed, to 1.3 or 1.4 or so.

Ok, most of those things were said before also by Hobbit Power, and I wanted to make the caves kind like ruins or something like that but well, it doesn't matter, I'll make that... It's always good to hear possitives reviews, and I'll do what you say, you know lot more about maps than me 😄

**Comment by LuismiInTheZone** [comment]
You got new files, should be ok

**Comment by LuismiInTheZone** [comment]
Ok now, this should work correctly, I tested and have no problems, you have around here the new version if you want to see it also 👍 

> Interesting map, looks cool. I'm not sure how enjoyable it is to climb mountains all day but we'll find out when it is played. I think you should add more ores to the dungeons, maybe not mese or diamonds but at least iron and coal (which ore is up to you though). I also suggest adding a full set of ladders (so no missing pieces like it is now) to the dungeon of the red base so it's as accessible as the one near the blue base.
> 
> Definitely add some items to initial_stuff. I suggested a pickaxe to mine the ores and also for digging tunnels _through_ the mountain. Torches always come in handy, especially since this is a rather dark map because of the skybox. Ladders to facilitate climbing the mountains.
> 
> EDIT: It could be a fun idea to slightly up the phys_speed, to 1.3 or 1.4 or so.

**Comment by MoreKat** [comment]
How many ores did you put on each side? I found 11 iron ores in blue's dungeon and only 7 in red's. I didn't check this with world edit, just dug them all up so it could be inaccurate which is why I'm asking. In general I think you can add some more to each side. Also, one last thing you need to fix is the screenshot's aspect ratio. Currently it's 750x382, it should be 750x500 (ratio 3:2)

**Comment by LuismiInTheZone** [comment]
> How many ores did you put on each side? I found 11 iron ores in blue's dungeon and only 7 in red's. I didn't check this with world edit, just dug them all up so it could be inaccurate which is why I'm asking. In general I think you can add some more to each side. Also, one last thing you need to fix is the screenshot's aspect ratio. Currently it's 750x382, it should be 750x500 (ratio 3:2)

I putted 10 of iron and coal at each team... I don't know if there's something default from the map...

**Comment by LuismiInTheZone** [comment]
https://user-images.githubusercontent.com/83732589/209076638-63c3deec-3495-46e6-8e58-c3e66dd46a0f.mov


https://user-images.githubusercontent.com/83732589/209076648-155e639b-8194-4e94-9ea3-74a855790297.mov

**Comment by LuismiInTheZone** [comment]
screenshot

**Comment by HobbitPower** [comment]
> How many ores did you put on each side? I found 11 iron ores in blue's dungeon and only 7 in red's. I didn't check this with world edit, just dug them all up so it could be inaccurate which is why I'm asking. In general I think you can add some more to each side. Also, one last thing you need to fix is the screenshot's aspect ratio. Currently it's 750x382, it should be 750x500 (ratio 3:2)

Checked with WE and there are 22 in total, 11 on each side. But adding more is a good idea.

However @LuismiInTheZone I've found some other problems:
* There are still breakable parts in the outer barrier. I thought in an older version you had fixed them. Now I'm again able to break some nodes (default:stone, default:sand etc). 
Screenshot 2022-12-22 105458
That torch btw wasn't placed by me btw: 
Screenshot 2022-12-22 105850
So it would be good to control the barriers again. 
* Speaking of barriers, the red barrier leaves holes after the game starts. If you want to fix that, now you still have the chance ^^
Screenshot 2022-12-22 105153
* Also there is an issue with your chests. I can't open them. Can you tell me if that's the case for you too? Tbh I haven't seen that before, so I will ask LV & Co, if that happens to others too. 
* Second thing about the chests: You may want to set 2 chestzones, one for each team. Just noticed that often one team has more chests than the other. 

Otherwise I like the map, bit of a challenge 👍

**Comment by ColorBlob** [comment]
This is a beautiful-looking map. The terrain is very interesting and It's going to be very interesting to capture (although I find this map _slightly_ similar to Green Hills). I did find 2 problems:
- On team blue, if you dig straight down during build time, you get teleported back to spawn.
- Chests are spawning inside almost-inaccessible caves. 

I have a few suggestions on the barrier placement:
- The red barrier still leaves holes after the match starts, as HobbitPower pointed out previously. That includes trees and terrain. It would be better to fill them up.
image
image
- There's a breakable layer outside the barrier of the map. That's okay, but the placement of the glass barrier and stone barrier is "chunky", and sometimes it creates giant walls and "glass indents", which is destroying the looks of the map.
image
image
- For some reason, one of the "chunky" barrier walls on blue's side included a big chunk of breakable stone, and I don't understand your intentions on that.
image
- Finally, there are random chunks of leaves floating near the barrier, and I recommend you to clean them.


The following are my personal opinion:
1. I don't have any trouble climbing the mountains, but it's going to be a nightmare for mobile players since it's pretty hard to press jump and sneak at the same time on mobile.
2. The place where the players from opposite teams are most likely to meet is on the mountain. But, the steep mountain terrain is horrible ground for PVP, and that's not going to be fun in my opinion.

**Comment by LuismiInTheZone** [comment]
> This is a beautiful-looking map. The terrain is very interesting and It's going to be very interesting to capture (although I find this map _slightly_ similar to Green Hills). I did find 2 problems:
> 
> * On team blue, if you dig straight down during build time, you get teleported back to spawn.
> * Chests are spawning inside almost-inaccessible caves.
> 
> I have a few suggestions on the barrier placement:
> 
> * The red barrier still leaves holes after the match starts, as HobbitPower pointed out previously. That includes trees and terrain. It would be better to fill them up.
>   image
>   image
> * There's a breakable layer outside the barrier of the map. That's okay, but the placement of the glass barrier and stone barrier is "chunky", and sometimes it creates giant walls and "glass indents", which is destroying the looks of the map.
>   image
>   image
> * For some reason, one of the "chunky" barrier walls on blue's side included a big chunk of breakable stone, and I don't understand your intentions on that.
>   image
> * Finally, there are random chunks of leaves floating near the barrier, and I recommend you to clean them.
> 
> The following are my personal opinion:
> 
> 1. I don't have any trouble climbing the mountains, but it's going to be a nightmare for mobile players since it's pretty hard to press jump and sneak at the same time on mobile.
> 2. The place where the players from opposite teams are most likely to meet is on the mountain. But, the steep mountain terrain is horrible ground for PVP, and that's not going to be fun in my opinion.

1. I can't spawn at team blue and I don't know why... So... I'll make that again
2.  I putted last chests to spawn only at caves inside the big mountain
3.  I forgot that... I think... The map is very big... I'll destroy some trees
4. Emm... I'll make that again
5.  I putted that because at it's moment I destroyed that part of the map, so I needed to put all that, later I didn't take it out
6.  Same as 3rd one

I can't do nothing with mobile player... And I wanted to make something new, I don't know if pvp is good or bad...

**Comment by ColorBlob** [comment]
Most of the issues are fixed now 🙂. **None of the issues I found now are serious.** 

But, I still noticed that you didn't fill up the holes that the red barrier leaves behind. Plus, not only are there floating trees near the red barrier, there are slices of floating leaves near the outer glass barrier too.

I also found an extra slice of terrain outside the barrier behind red's base.
That is exactly where the "slices of trees" are.

That torch btw wasn't placed by me btw:
image

Also, I found a nice window:
image

**Comment by LuismiInTheZone** [comment]
> Most of the issues are fixed now :) **None of the issues I found now are serious.**
> 
> But, I still noticed that you didn't fill up the holes that the red barrier leaves behind. The trees still have holes in them, despite the fact that you have added tree trunks; and there are slices of floating leaves near the outer glass barrier too.
> 
> I also found an extra slice of terrain outside the barrier behind red's base. That is exactly where the "slices of trees" are.
> 
> That torch btw wasn't placed by me btw: image
> 
> Also, I found a nice window: image

I'm going to try re-add the map to see if chests can be open (and also fix those one hundred tiny problems...)

**Comment by ColorBlob** [comment]
So far I did not find any more bugs and issues. The barrier placements are really messed up and weird, but It's not causing any issues with the gameplay.

**Comment by HobbitPower** [comment]
Okay. I didn't find any other bugs, the chests seem to work. I still think it would be good if LandarVargan takes another look at the conf file before the map is going to be merged. (As a note to the other Map Managers).

The map itself is more challenging than other maps in my opinion, but that's to my taste. It could well need to be tweaked again if it is too hard to conquer or takes too long though. You could then add tunnels, for example, etc. 

But I for one would wait for the feedback of the players and give the map a chance. Then will we see if the map is fun to play.

**Pull request merged**